src/scheduler/server_info.c

```diff
@@ -3912,6 +3918,40 @@ create_resource_assn_for_node(node_info *ninfo)
 		}
 	}
 
+	/* Lastly if restrict_res_to_release_on_suspend is set, suspended jobs may not have released all their resources
+	 * This is tricky since a suspended job knows what resources they released.
+	 * We need to know what they didn't release to account for in the nodes resources_assigned
+	 */
+	if (ninfo->num_susp_jobs > 0) {
+		int i;
+		server_info *sinfo = ninfo->server;
+		for (i = 0; sinfo->jobs[i] != NULL; i++) {
+			if (sinfo->jobs[i]->job->is_suspended) {
+				nspec *ens;
+				ens = find_nspec(sinfo->jobs[i]->nspec_arr, ninfo);
+				if (ens != NULL) {
+					nspec *rns;
+					rns = find_nspec(sinfo->jobs[i]->job->resreleased, ninfo);
+					if (rns != NULL) {
+						resource_req *cur_req;
+						for (cur_req = ens->resreq; cur_req != NULL; cur_req = cur_req->next) {
+							if (cur_req->type.is_consumable)
+								if (find_resource_req(rns->resreq, cur_req->def) == NULL) {
```

Comment:
Can you please add a comment that we are trying to find resources which were not released and assigning them to node.

Reply:
I had one originally, but I removed it when I added the comment ahead of this block of code.  It already says that we need to find resources that are not released.  I don't want to have two comments that say the same thing.  If the previous comment doesn't explain this, please provide a suggestion on how to make it more clear.